lib/expression/node/OperatorNode.js

```diff
@@ -269,6 +269,29 @@ function factory (type, config, load, typed, math) {
       }
 
       return [lhsParens, rhsParens];
+    } else if ((args.length > 2) && ((root.getIdentifier() === 'OperatorNode:add') || (root.getIdentifier() === 'OperatorNode:multiply'))) {
+      var parensArray = [];
+
+      args.forEach(function (arg) {
+        var parens;
+        var argPrecedence = operators.getPrecedence(arg, parenthesis);
+        var assocWithArg = operators.isAssociativeWith(root, arg, parenthesis);
+        var argAssociativity = operators.getAssociativity(arg, parenthesis);
+        if (argPrecedence === null) {
+          //if the argument has no defined precedence, no parens are needed
+          parens = false;
+        } else if ((precedence === argPrecedence) && (associativity === argAssociativity) && !assocWithArg) {
+          parens = true;
+        } else if (argPrecedence < precedence) {
+          parens = true;
+        } else {
+          parens = false;
+        }
+
+        parensArray.push(parens);
```

Comment:
I wasn't too familiar with `Array.prototype.map`. I'll take a look at it tomorrow.